src/prefect/blocks/core.py

```diff
@@ -323,6 +325,31 @@ def __dispatch_key__(cls):
             return None  # The base class is abstract
         return block_schema_to_key(cls._to_block_schema())
 
+    @model_serializer(mode="wrap")
+    def ser_model(self, handler: Callable, info: SerializationInfo) -> Any:
+        jsonable_self = handler(self)
+        if (ctx := info.context) and ctx.get("include_secrets") is True:
+            jsonable_self.update(
+                {
+                    field_name: visit_collection(
+                        expr=getattr(self, field_name),
+                        visit_fn=partial(handle_secret_render, context=ctx),
+                        return_data=True,
+                    )
+                    for field_name in self.model_fields
+                }
+            )
+        if extra_fields := {
+            "block_type_slug": self.get_block_type_slug(),
+            "_block_document_id": self._block_document_id,
+            "_block_document_name": self._block_document_name,
+            "_is_anonymous": self._is_anonymous,
+        }:
+            jsonable_self |= {
+                key: value for key, value in extra_fields.items() if value is not None
+            }
+        return jsonable_self
```

Comment:
this is how we now reveal / redact (latter being the default) secrets for blocks when serializing

`handle_secret_render` at this time essentially just checks for `hasattr(value, 'get_secret_value')` on each item of the collection and returns the revealed / obscured value based on the serialization `context`